src/runtime_src/core/common/xclbin_parser.cpp

```diff
@@ -540,6 +540,54 @@ get_cus(const ip_layout* ip_layout, const std::string& kname)
   return ips;
 }
 
+std::map<std::string, std::vector<instance_object>>
+get_kernel_instance_map( const char* xml_data, size_t xml_size) {
```

Comment:
This is xrt::xclbin::get_ips() and variants.  The xrt::xclbin::ip has member functions for accessing most of the data you need,  if some data is missing, please let @stsoe know.